packages/apptive_grid_form/lib/src/apptive_grid_form.dart

```diff
@@ -343,8 +343,24 @@ class ApptiveGridFormDataState extends State<ApptiveGridFormData> {
     return ApptiveGridLocalization(
       child: Builder(
         builder: (buildContext) {
+          final hasInvalidTextFields = _formData?.components
+                  ?.where(
+                    (element) =>
+                        element.field.type == DataType.text &&
+                        !element.options.multi &&
+                        element.data.value?.contains('\n') == true,
+                  )
+                  .isNotEmpty ==
+              true;
+          if (_error == null && hasInvalidTextFields) {
+            final localization = ApptiveGridLocalization.of(buildContext)!;
+            _error = localization.multilineTextOverflowError;
+          }
```

Comment:
This check should rather happen in the loading callback instead of the build method to prevent having to do it for every build

Reply:
moved to parsing